### PR TITLE
Add copilot instructions and issue template tweaks

### PR DESCRIPTION
**Pull request by wandergeek**
These will be used by the copilot agent to work on issues related to this repo.

### CODE REVIEW

File: .github/copilot-instructions.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+This is a Go based repository hosting a Terrform provider for the elastic stack (elasticsearch and kibana) APIs.  This repo currently supports both [plugin framework](https://developer.hashicorp.com/terraform/plugin/framework/getting-started/code-walkthrough) and [sdkv2](https://developer.hashicorp.com/terraform/plugin/sdkv2) resources. All new resources _must_ use the plugin framework.
```
Comment (tobio) [inline .github/copilot-instructions.md]:
```suggestion
This is a Go based repository hosting a Terrform provider for the elastic stack (elasticsearch and kibana) APIs.  This repo currently supports both [plugin framework](https://developer.hashicorp.com/terraform/plugin/framework/getting-started/code-walkthrough) and [sdkv2](https://developer.hashicorp.com/terraform/plugin/sdkv2) resources. Unless you're told otherwise, all new resources _must_ use the plugin framework. 
```

The data source we worked on together was an SDK resource, it's a bit of a special case, but I think it's worth being open to that.

File: .github/copilot-instructions.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+### Development Flow
+- Develop feature or fix bug
+- Write tests to validate behavior
+- Run `make test` to run test suite
+
+## Repository Structure
+
+• **docs/** - Documentation files
+  • **data-sources/** - Documentation for Terraform data sources (51 files)
```
Comment (tobio) [inline .github/copilot-instructions.md]: Why the file counts in this listing? It will almost immediately out of date.

File: .github/copilot-instructions.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+For further information, please see [README.md](../README.md) and the [CONTRIBUTING.md](../CONTRIBUTING.md) docs.
+
+## Code Standards
+
+### Required Before Each Commit
+- Run `make fmt` before committing any changes to ensure proper code formatting, this will run gofmt on all Go files to maintain consistent style.
+- Run `make lint` to ensure any linting errors have not surfaced with your changes
+
+### Required Before Pull Request
+- Run `make gen` to update the documentation and code based on your changes.
```
Comment (tobio) [inline .github/copilot-instructions.md]:
```suggestion
- Run `make docs-generate` to update the documentation.
```

`make gen` is probably going to do things we don't want right now. Updating the Kibana client should be something we can support, but it's not there yet.

File: .github/copilot-instructions.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+• **templates/** - Template files for documentation generation
+  • **data-sources/** - Data source documentation templates
+  • **resources/** - Resource documentation templates
+  • **guides/** - Guide documentation templates
+• **xpprovider/** - Additional provider functionality needed for Crossplane
+
+## Key Guidelines
+* Follow Go best practices and idiomatic patterns
+* Maintain existing code structure and organization
+* Write unit tests for new functionality. Use table-driven unit tests when possible.
```
Comment (tobio) [inline .github/copilot-instructions.md:118]:
```suggestion
* Write unit tests for new functionality. Use table-driven unit tests when possible.
* When creating a new Plugin Framework based resource, follow the code organisation of `internal/elasticsearch/security/system_user` 
```

File: .github/copilot-instructions.md (hunk 2 above)
Comment (wandergeek) [reply .github/copilot-instructions.md]: lol, because I crossed the streams and put AI in the AI

File: .github/copilot-instructions.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+- Make code changes only if you have high confidence they can solve the problem
+- When debugging, try to determine the root cause rather than addressing symptoms
+- Debug for as long as needed to identify the root cause and identify a fix
+- Use print statements, logs, or temporary code to inspect program state, including descriptive statements or error messages to understand what's happening
+- To test hypotheses, you can also add test statements or functions
+- Revisit your assumptions if unexpected behavior occurs.
+- You MUST iterate and keep going until the problem is solved.
+
+## 6. Testing
+- Run tests frequently using `make test`
```
Comment (tobio) [inline .github/copilot-instructions.md]:
```suggestion
- Run tests frequently using `make test` and `make testacc`
```

We _really_ care about the acceptance tests, unit tests are kind of a coding tool.

File: .github/copilot-instructions.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+- If tests fail, analyze failures and revise your patch.
+- Write additional tests if needed to capture important behaviors or edge cases.
+- Ensure all tests pass before finalizing.
+
+## 7. Final Verification
+- Confirm the root cause is fixed.
+- Review your solution for logic correctness and robustness.
+- Iterate until you are extremely confident the fix is complete and all tests pass.
+- Run `make lint` to ensure any linting errors have not surfaced with your changes
+- Run `make fmt` before committing any changes to ensure proper code formatting, this will run gofmt on all Go files to maintain consistent style.
```
Comment (tobio) [inline .github/copilot-instructions.md]:
```suggestion
- Run `make lint` to ensure any linting errors have not surfaced with your changes. This task may automatically correct any linting errors, and regenerate documentation. Include any changes in your commit. 
```


`make fmt` is called by `make lint`.

File: .github/copilot-instructions.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+## 8. Final Reflection and Additional Testing
+- Reflect carefully on the original intent of the user and the problem statement.
+- Think about potential edge cases or scenarios that may not be covered by existing tests.
+- Write additional tests that would need to pass to fully validate the correctness of your solution.
+- Run these new tests and ensure they all pass.
+- Be aware that there are additional hidden tests that must also pass for the solution to be successful.
```
Comment (tobio) [inline .github/copilot-instructions.md:73]: Why are there hidden tests? Why can't we just say make sure `make testacc` and `make test` are successful?

Comment (wandergeek) [reply .github/copilot-instructions.md:73]: It forces it to further consider edge cases 🤫

**Review by tobio — APPROVED** [review]
Lets give it a crack
